taskcluster/kinds/merge-devset/kind.yml

```diff
@@ -12,18 +12,10 @@ transforms:
     - translations_taskgraph.transforms.cached_tasks:transforms
     - taskgraph.transforms.task:transforms
 
+# todo: why bicleaner and clean-corpus are required here? it doesn't work without them
```

Comment:
What is the issue?
I bet @bhearsum or @gabrielBusta can help

Reply:
I'd have to know more about what doesn't work to say anything with certainty, but in general you need to have `kind-dependencies` set if any tasks in this kind are generated from another kind.

(I am a little bit surprised that `clean-corpus` is required here....that looks like it may be an historical artifact from when all of the `merge` tasks were in the same kind...)

Reply:
merging devest does not depend on cleaning or bicleaner. We merge raw datasets here. When I remove those two and validate the taskgraph it says:
```
2024-01-22 15:52:17,682 - ERROR - Error loading tasks for kind merge-devset:
Traceback (most recent call last):
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/generator.py", line 311, in _run
    new_tasks = kind.load_tasks(
                ^^^^^^^^^^^^^^^^
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/generator.py", line 76, in load_tasks
    tasks = [
            ^
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/generator.py", line 76, in <listcomp>
    tasks = [
            ^
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/task.py", line 1347, in check_run_task_caches
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/task.py", line 1276, in check_task_dependencies
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/task.py", line 1262, in check_task_identifiers
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/task.py", line 1243, in chain_of_trust
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/task.py", line 1236, in add_github_checks
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/task.py", line 1061, in build_task
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/task.py", line 1029, in add_index_routes
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/task.py", line 957, in process_treeherder_metadata
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/task.py", line 919, in validate
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/task.py", line 907, in task_name_from_label
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/task.py", line 863, in set_defaults
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/task.py", line 842, in set_implementation
    for task in tasks:
  File "/Users/epavlov/mozilla/source/origin/firefox-translations-training/taskcluster/translations_taskgraph/transforms/cached_tasks.py", line 117, in cache_task
    for task in order_tasks(config, tasks):
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/cached_tasks.py", line 22, in order_tasks
    pending = deque(tasks)
              ^^^^^^^^^^^^
  File "/Users/epavlov/mozilla/source/origin/firefox-translations-training/taskcluster/translations_taskgraph/transforms/cached_tasks.py", line 69, in add_cache
    for job in jobs:
  File "/Users/epavlov/mozilla/source/origin/firefox-translations-training/taskcluster/translations_taskgraph/transforms/cached_tasks.py", line 55, in resolved_keyed_by_fields
    for job in jobs:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/base.py", line 144, in __call__
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/run/__init__.py", line 361, in make_task_description
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/run/__init__.py", line 236, in use_fetches
    for task in order_tasks(config, tasks):
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/cached_tasks.py", line 22, in order_tasks
    pending = deque(tasks)
              ^^^^^^^^^^^^
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/run/__init__.py", line 164, in add_resource_monitor
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/run/__init__.py", line 152, in set_label
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/run/__init__.py", line 136, in set_implementation
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/run/__init__.py", line 116, in rewrite_when_to_optimization
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/base.py", line 144, in __call__
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/task_context.py", line 85, in render_task
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/base.py", line 144, in __call__
    for task in tasks:
  File "/Users/epavlov/mozilla/source/origin/firefox-translations-training/taskcluster/translations_taskgraph/transforms/find_upstreams.py", line 97, in upstreams_for_locales
    for job in jobs:
  File "/Users/epavlov/mozilla/source/origin/firefox-translations-training/taskcluster/translations_taskgraph/transforms/find_upstreams.py", line 82, in resolve_keyed_by_fields
    cleaning_type = get_cleaning_type(config.kind_dependencies_tasks.values())
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/epavlov/mozilla/source/origin/firefox-translations-training/taskcluster/translations_taskgraph/transforms/find_upstreams.py", line 75, in get_cleaning_type
    raise Exception("Unable to find cleaning type!")
Exception: Unable to find cleaning type!
Traceback (most recent call last):
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/main.py", line 907, in main
    return args.command(vars(args))
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/main.py", line 430, in show_taskgraph
    ret = generate_taskgraph(options, parameters, logdir)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/main.py", line 192, in generate_taskgraph
    out = format_taskgraph(options, spec, logfile(spec))
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/main.py", line 148, in format_taskgraph
    tg = getattr(tgg, options["graph_attr"])
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/generator.py", line 170, in full_task_graph
    return self._run_until("full_task_graph")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/generator.py", line 425, in _run_until
    k, v = next(self._run)
           ^^^^^^^^^^^^^^^
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/generator.py", line 311, in _run
    new_tasks = kind.load_tasks(
                ^^^^^^^^^^^^^^^^
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/generator.py", line 76, in load_tasks
    tasks = [
            ^
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/generator.py", line 76, in <listcomp>
    tasks = [
            ^
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/task.py", line 1347, in check_run_task_caches
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/task.py", line 1276, in check_task_dependencies
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/task.py", line 1262, in check_task_identifiers
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/task.py", line 1243, in chain_of_trust
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/task.py", line 1236, in add_github_checks
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/task.py", line 1061, in build_task
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/task.py", line 1029, in add_index_routes
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/task.py", line 957, in process_treeherder_metadata
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/task.py", line 919, in validate
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/task.py", line 907, in task_name_from_label
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/task.py", line 863, in set_defaults
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/task.py", line 842, in set_implementation
    for task in tasks:
  File "/Users/epavlov/mozilla/source/origin/firefox-translations-training/taskcluster/translations_taskgraph/transforms/cached_tasks.py", line 117, in cache_task
    for task in order_tasks(config, tasks):
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/cached_tasks.py", line 22, in order_tasks
    pending = deque(tasks)
              ^^^^^^^^^^^^
  File "/Users/epavlov/mozilla/source/origin/firefox-translations-training/taskcluster/translations_taskgraph/transforms/cached_tasks.py", line 69, in add_cache
    for job in jobs:
  File "/Users/epavlov/mozilla/source/origin/firefox-translations-training/taskcluster/translations_taskgraph/transforms/cached_tasks.py", line 55, in resolved_keyed_by_fields
    for job in jobs:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/base.py", line 144, in __call__
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/run/__init__.py", line 361, in make_task_description
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/run/__init__.py", line 236, in use_fetches
    for task in order_tasks(config, tasks):
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/cached_tasks.py", line 22, in order_tasks
    pending = deque(tasks)
              ^^^^^^^^^^^^
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/run/__init__.py", line 164, in add_resource_monitor
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/run/__init__.py", line 152, in set_label
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/run/__init__.py", line 136, in set_implementation
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/run/__init__.py", line 116, in rewrite_when_to_optimization
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/base.py", line 144, in __call__
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/task_context.py", line 85, in render_task
    for task in tasks:
  File "/Users/epavlov/Library/Caches/pypoetry/virtualenvs/firefox-translations-training-U0HVfMmK-py3.11/lib/python3.11/site-packages/taskgraph/transforms/base.py", line 144, in __call__
    for task in tasks:
  File "/Users/epavlov/mozilla/source/origin/firefox-translations-training/taskcluster/translations_taskgraph/transforms/find_upstreams.py", line 97, in upstreams_for_locales
    for job in jobs:
  File "/Users/epavlov/mozilla/source/origin/firefox-translations-training/taskcluster/translations_taskgraph/transforms/find_upstreams.py", line 82, in resolve_keyed_by_fields
    cleaning_type = get_cleaning_type(config.kind_dependencies_tasks.values())
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/epavlov/mozilla/source/origin/firefox-translations-training/taskcluster/translations_taskgraph/transforms/find_upstreams.py", line 75, in get_cleaning_type
    raise Exception("Unable to find cleaning type!")
Exception: Unable to find cleaning type!

```

Reply:
this issue is not related to this PR though

Reply:
Ah, I see. Yep - that's an oversight from when `merge-devset` moved to its own kind. https://github.com/mozilla/firefox-translations-training/pull/385 fixes it.